### ISSUE TITLE
Is the function `get_attn_mask`  the same one used in the paper on arxiv?

### ISSUE THREAD

**Issue by azfarkhoja305** [VITA-Group/TransGAN#20]
Hi,
When I try to recreate the attention mask using the function `get_attn_mask`  in `models/TransGAN_8_8_G2_1.py` or `models/TransGAN_8_8_1.py`

I end up with the image below.
The image on the left is the output from `get_attn_mask` with `N=32*32`, `w = 25`.
The image on the right is reshaping the mask for 495th row (or pixel) into into a 32x32 image (like in the paper).
The grey color represents that pixel. 

As it's visible the outputs won't match up with the paper for any value of `w`. 

I appreciate any help that you can provide. Thanks!

![download (1)](https://user-images.githubusercontent.com/36442229/116124044-c4f82e80-a691-11eb-9682-50136f4732cb.png)

(PS: Added the image from the paper for reference.)
![paper](https://user-images.githubusercontent.com/36442229/116130988-d3e2df00-a699-11eb-9457-7708bd2f8084.PNG)

**Comment by yueruchen** [comment]
The figure just shows an expected attentive value of given pixels while the mask does not actually match the figure (You will find that the figure even is not symmetric.) If you want a 2D structure mask you can slightly change the mask function. But due to my previous experience, the performance is nearly same with 1D structure mask. I guess the reason is that both of them will finally shrink to the standard self-attention during the training process.

**Comment by azfarkhoja305** [comment]
Thanks for the prompt reply. 
It's just that I find the image a bit misleading since even with the biggest window size `w=14` (as described in the footnotes 1 of pg 6 of the paper), it will be attending to pixels in a straight horizontal line.

Anyways, Great work on the paper, enjoyed recreating it.
